app_integrations/apps/onelogin.py

```diff
@@ -0,0 +1,192 @@
+"""
+Copyright 2017-present, Airbnb Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+   http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+"""
+from datetime import datetime
+import re
+import requests
+
+from app_integrations import LOGGER
+from app_integrations.apps.app_base import AppIntegration
+
+
+class OneLoginApp(AppIntegration):
+    """OneLogin StreamAlert App"""
+    _ONELOGIN_EVENTS_URL = 'https://api.us.onelogin.com/api/1/events'
+    _ONELOGIN_TOKEN_URL = 'https://api.us.onelogin.com/auth/oauth2/v2/token'
+    # OneLogin API returns 50 events per page
+    _MAX_EVENTS_LIMIT = 50
+
+    # Define our authorization headers variable
+    def __init__(self, config):
+        super(OneLoginApp, self).__init__(config)
+        self._auth_headers = None
+        self._next_page_url = None
+
+    @classmethod
+    def _type(cls):
+        return 'events'
+
+    @classmethod
+    def _endpoint(cls):
+        """Class method to return the OneLogin events endpoint
+
+        Returns:
+            str: Path of the events endpoint to query
+        """
+        return cls._ONELOGIN_EVENTS_URL
+
+    @classmethod
+    def service(cls):
+        return 'onelogin'
+
+    def _generate_headers(self, token_url, client_secret, client_id):
+        """Each request will request a new token to call the resources APIs.
+
+        More details to be found here:
+            https://developers.onelogin.com/api-docs/1/oauth20-tokens/generate-tokens-2
+
+        Returns:
+            str: Bearer token to be used to call the OneLogin resource APIs
+        """
+        authorization = 'client_id: %s, client_secret: %s' % (client_id, client_secret)
+        headers_token = {'Authorization': authorization,
+                         'Content-Type': 'application/json'}
+
+        response = requests.post(token_url,
```

Comment:
I anticipate potentially needing `post` abilities in other future apps as well. Could we move this functionality to the `AppIntegration` base class in `app_base.py` with a method name of something like `_make_post_request` and change the `_make_request` method to `_make_get_request`? By doing this, the new method could just return False or the JSON from the response (similar to how `_make_request` works now).

If this is too much right now, this can happen at a later date - just a thought!

Reply:
Should I also refactor the duo application to use the new `_make_get_requests` in this PR?

Reply:
Yes please! :)